src/main/java/hudson/plugins/ansicolor/AnsiHtmlOutputStream.java

```diff
@@ -627,4 +627,33 @@ protected void processDefaultTextColor() throws IOException {
     protected void processDefaultBackgroundColor() throws IOException {
         setBackgroundColor(null);
     }
+    
+    @Override
+    protected void processEraseLine(int eraseOption) throws IOException {
+        emitNothing();
+    }
+
+    @Override
+    protected void processCursorDown(int count) throws IOException {
+        emitNothing();
+    }
+
+    @Override
+    protected void processCursorUp(int count) throws IOException {
+        emitNothing();
+    }
+
+    @Override
+    protected void processCursorLeft(int count) throws IOException {
+        emitNothing();
+    }
+
+    @Override
+    protected void processCursorUpLine(int count) throws IOException {
+        emitNothing();
+    }
+
+    private void emitNothing() {
+        emitter.emitHtml("");
+    }
```

Comment:
Instead of adding this method here, I would rename [`ColorConsoleWrapper$EmitterImpl.emitRedundantReset()`](https://github.com/jenkinsci/ansicolor-plugin/blob/bb959c39fdfeef15f52c876df3a361c7468fef6c/src/main/java/hudson/plugins/ansicolor/ColorConsoleAnnotator.java#L105-L108) to `emitInvisibleAnsiSequence` or something similar and call it in the body of the methods here instead of adding a helper method. That that method emits `null` instead of `""`, which has [special handling](https://github.com/jenkinsci/ansicolor-plugin/blob/bb959c39fdfeef15f52c876df3a361c7468fef6c/src/main/java/hudson/plugins/ansicolor/ColorConsoleAnnotator.java#L89) inside of `emitHtml`, so as long as it works, I think it would be better to emit `null` to make it clear that nothing is actually being emitted.

Reply:
> 
> 
> Instead of adding this method here, I would rename [`ColorConsoleWrapper$EmitterImpl.emitRedundantReset()`](https://github.com/jenkinsci/ansicolor-plugin/blob/bb959c39fdfeef15f52c876df3a361c7468fef6c/src/main/java/hudson/plugins/ansicolor/ColorConsoleAnnotator.java#L105-L108) to `emitInvisibleAnsiSequence` or something similar and call it in the body of the methods here instead of adding a helper method.

I have nothing against this part. 

> 
> 
> That method emits `null` instead of `""`, which has [special handling](https://github.com/jenkinsci/ansicolor-plugin/blob/bb959c39fdfeef15f52c876df3a361c7468fef6c/src/main/java/hudson/plugins/ansicolor/ColorConsoleAnnotator.java#L89) inside of `emitHtml`, so as long as it works, I think it would be better to emit `null` to make it clear that nothing is actually being emitted.

But I can't say that I agree with

> better to emit `null`

Explicitly providing a `null` as a method parameter is just a `NPE` waiting to happen. I specifically avoided calling `ColorConsoleWrapper$EmitterImpl.emitRedundantReset()` for this exact reason. Maybe a cleaner approach would be to define a private constant inside of `ColorConsoleWrapper$EmitterImpl`
```java
private static final String NO_CORR_TAGS = "";
```
use it in `ColorConsoleWrapper$EmitterImpl.emitRedundantReset()`
and then check for it in the condition
```java
if (!NO_CORR_TAGS.equals(html)) {
// ...
}
```

Reply:
> Explicitly providing a null as a method parameter is just a NPE waiting to happen

You could add `@CheckForNull` to the `String html` parameter in `EmitterImpl.emitHtml` to get SpotBugs to warn about any issues if you are concerned about someone introducing an unsafe change, but we already have tests that pass `null` via `emitRedundantReset`, so if someone did make a change that didn't handle `null` correctly, the tests should start failing, and the issue would have to be fixed before the PR was merged.

If you really don't like that, I think it would work fine to switch `emitRedundantReset` to use `""`, and change the logic in `emitHtml` to compare `html` against `""` where it currently compares `html` to `null`, but that seems riskier in the sense that someone might break things in the future in a subtle way that goes unnoticed at first because `""` is a valid `String`.